src/supernode/healthcheckapi.cpp

```diff
@@ -0,0 +1,53 @@
+#include "healthcheckapi.h"
+#include "rpc/core_rpc_server_commands_defs.h"
+#include "storages/http_abstract_invoke.h"
+#include "net/jsonrpc_structs.h"
+#include "net/net_utils_base.h"
+
+static const std::string HEALTH_URI("/health");
+
+struct HealthResponse {
+    string NodeAccess;
+
+    BEGIN_KV_SERIALIZE_MAP()
+        KV_SERIALIZE(NodeAccess)
+    END_KV_SERIALIZE_MAP()
+
+};
+
+supernode::HealthcheckAPI::HealthcheckAPI(const string &daemonAddress)
+{
+    boost::optional<epee::net_utils::http::login> login{};
+    m_http_client.set_server("http://" + daemonAddress, login);
+}
+
+bool supernode::HealthcheckAPI::processHealthchecks(const string uri, epee::net_utils::http::http_response_info &response_info)
+{
+    if (uri == HEALTH_URI)
+    {
+        HealthResponse response;
+        response.NodeAccess = cryptonodeCheck() ? "OK" : "Fail";
+        epee::serialization::store_t_to_json(response, response_info.m_body);
```

Comment:
what's .m_boady? empty?

Reply:
m_body is body of response
yes, at the start it is empty, epee::serialization::store_t_to_json() will write response structure in it.